go/libraries/doltcore/merge/constraint_violations.go

```diff
@@ -0,0 +1,507 @@
+		if err != nil {
+			return nil, err
+		}
+		postParentIndex, ok := postParentSch.Indexes().GetByNameCaseInsensitive(foreignKey.ReferencedTableIndex)
+		if !ok {
+			return nil, fmt.Errorf("foreign key %s should have index %s on table %s but it cannot be found",
+				foreignKey.Name, foreignKey.ReferencedTableIndex, foreignKey.ReferencedTableName)
+		}
+		postParentIndexData, err := postParentTable.GetIndexRowData(ctx, postParentIndex.Name())
+		if err != nil {
+			return nil, err
+		}
+
+		postChildTable, childTblName, ok, err := newRoot.GetTableInsensitive(ctx, foreignKey.TableName)
+		if err != nil {
+			return nil, err
+		}
+		if !ok {
+			return nil, doltdb.ErrTableNotFound
+		}
+		postChildSch, err := postChildTable.GetSchema(ctx)
+		if err != nil {
+			return nil, err
+		}
+		postChildData, err := postChildTable.GetRowData(ctx)
+		if err != nil {
+			return nil, err
+		}
+		postChildIndex, ok := postChildSch.Indexes().GetByNameCaseInsensitive(foreignKey.TableIndex)
+		if !ok {
+			return nil, fmt.Errorf("foreign key %s should have index %s on table %s but it cannot be found",
+				foreignKey.Name, foreignKey.TableIndex, childTblName)
+		}
+		postChildIndexData, err := postChildTable.GetIndexRowData(ctx, postChildIndex.Name())
+		if err != nil {
+			return nil, err
+		}
+
+		preParentTable, _, ok, err := ourRoot.GetTableInsensitive(ctx, foreignKey.ReferencedTableName)
+		if err != nil {
+			return nil, err
+		}
+		if ok {
+			preParentSch, err := preParentTable.GetSchema(ctx)
+			if err != nil {
+				return nil, err
+			}
+			preParentData, err := preParentTable.GetRowData(ctx)
+			if err != nil {
+				return nil, err
+			}
+			postChildTable, err = parentFkConstraintViolations(
+				ctx,
+				foreignKey,
+				postChildTable,
+				preParentSch, postParentSch, postChildSch,
+				postParentIndex, postChildIndex,
+				preParentData, postParentData, postChildData, postParentIndexData, postChildIndexData)
+			if err != nil {
+				return nil, err
+			}
+			postChildData, err = postChildTable.GetRowData(ctx)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		preChildTable, _, ok, err := ourRoot.GetTableInsensitive(ctx, foreignKey.TableName)
+		if err != nil {
+			return nil, err
+		}
+		if ok {
+			preChildSch, err := preChildTable.GetSchema(ctx)
+			if err != nil {
+				return nil, err
+			}
+			preChildData, err := preChildTable.GetRowData(ctx)
+			if err != nil {
+				return nil, err
+			}
+			postChildTable, err = childFkConstraintViolations(
+				ctx,
+				foreignKey,
+				postChildTable,
+				preChildSch, postChildSch,
+				postParentIndex, postChildIndex,
+				postParentIndexData, preChildData, postChildData)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		newRoot, err = newRoot.PutTable(ctx, childTblName, postChildTable)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return newRoot, nil
+}
+
+//TODO: stolen from IndexEditor, need to move the original somewhere sensible before merging
```

Comment:
Add a method to schema.Index so you can just do index.ToTableTuple(nbf, indexKey) or something?